### PR TITLE
Get phpstan build passing at level 2.

### PR DESCRIPTION
**Pull request by ADmad**
Aiming for level 5 was a bit over ambitious at this point. Also having the build under "allow failures" means new errors could be overlooked. Bumping the levels sequentially would be a better approach. Up to level 2 now vs. level 1 on master :muscle:.

### CODE REVIEW

File: src/Network/Socket.php [hunk 1]
```diff
@@ -315,7 +315,7 @@ public function lastError()
     /**
      * Set the last error.
      *
-     * @param int $errNum Error code
+     * @param int|null $errNum Error code
```
Comment (dereuromark) [inline src/Network/Socket.php]: Is that true? Or should instead the calling code better cast?

File: src/View/View.php [hunk 2]
```diff
@@ -648,7 +648,7 @@ public function setLayout($name)
      *
      * @deprecated 3.5.0 Use getLayout()/setLayout() instead.
      * @param string|null $name Layout file name to set. If null returns current name.
-     * @return string|null
+     * @return string|false
```
Comment (dereuromark) [inline src/View/View.php]: It can still be null (if not yet set), right?

Comment (ADmad) [reply src/View/View.php]: No, the property is set to `'default'` during declaration.

Comment (dereuromark) [reply src/View/View.php]: You are right, and setTemplate($name) strictly speaking does not accept to make it null again.

File: src/View/Helper/NumberHelper.php [hunk 3]
```diff
@@ -152,7 +152,7 @@ public function format($number, array $options = [])
         $formatted = $this->_engine->format($number, $options);
         $options += ['escape' => true];
 
-        return $options['escape'] ? h($formatted) : $formatted;
+        return $options['escape'] ? (string)h($formatted) : $formatted;
```
Comment (lorenzo) [inline src/View/Helper/NumberHelper.php]: Why was this cast necessary? Seems wasteful

Comment (dereuromark) [reply src/View/Helper/NumberHelper.php]: Because h() being a god function.. Both string and array input output.

Comment (ADmad) [reply src/View/Helper/NumberHelper.php]: `h()` can return a string or array, hence without the cast phpstan is unhappy.

Perhaps in 4.x we can make it just accept and return strings. I doubt passing array to it is commonly used.

**Comment by lorenzo** [comment]
Do you think this may be a perf regression? This is the type of functions that get called very often in a loop

**Comment by ADmad** [comment]
@lorenzo I removed the string casing in 27537e55c32503957e67d0f5098ec8e44823a645. The earlier `string|array` return type was a lie anyway since the method has this:
```php
        } elseif (is_bool($text) || is_null($text) || is_int($text)) {
            return $text;
        }
``

Mistakenly directly commit to 4.x instead of making a PR.

**Comment by lorenzo** [comment]
Great!

**Pull request merged**